src/plugins/discover/public/application/main/components/top_nav/get_top_nav_links.tsx

```diff
@@ -159,17 +157,49 @@ export const getTopNavLinks = ({
     }),
     testId: 'shareTopNavButton',
     run: async (anchorElement: HTMLElement) => {
-      const updatedDataView = await persistDataView(dataView);
-      if (!services.share || !updatedDataView) {
-        return;
-      }
-      const sharingData = await getSharingData(searchSource, state.appState.getState(), services);
+      if (!services.share) return;
 
+      const { locator } = services;
+      const appState = state.appState.getState();
+      const { timefilter } = services.data.query.timefilter;
+      const timeRange = timefilter.getTime();
+      const refreshInterval = timefilter.getRefreshInterval();
+      const { grid, ...otherState } = appState;
+      const filters = services.filterManager.getFilters();
+
+      // Share -> Get links -> Snapshot
+      const params: DiscoverAppLocatorParams = {
+        ...otherState,
+        ...(savedSearch.id ? { savedSearchId: savedSearch.id } : {}),
+        ...(dataView?.isPersisted()
+          ? { dataViewId: dataView?.id }
+          : { dataViewSpec: dataView?.toSpec() }),
+        filters,
+        timeRange,
+        refreshInterval,
+      };
+      const relativeUrl = locator.getRedirectUrl(params);
+
+      // This logic is duplicated from `relativeToAbsolute` (for bundle size reasons). Ultimately, this should be
+      // replaced when https://github.com/elastic/kibana/issues/153323 is implemented.
+      const link = document.createElement('a');
+      link.setAttribute('href', relativeUrl);
+      const shareableUrl = link.href;
```

Comment:
I think this needs to be updated, since this was merged, but doesn't seem to solve the issue

https://github.com/elastic/kibana/issues/153323#issuecomment-1528021521

I do agree with you, it should not be necessary to use this workaround. If getUrl works it should also be possible to use `getRedirectUrl`as an absolute URL .. FYI @vadimkibana

Reply:
Anyway thx for taking care of the bundle size @lukasolson